src/journal/journald-syslog.c

```diff
@@ -311,40 +313,49 @@ static void syslog_skip_date(char **buf) {
 void server_process_syslog_message(
         Server *s,
         const char *buf,
+        size_t buf_len,
         const struct ucred *ucred,
         const struct timeval *tv,
         const char *label,
         size_t label_len) {
 
         char syslog_priority[sizeof("PRIORITY=") + DECIMAL_STR_MAX(int)],
-             syslog_facility[sizeof("SYSLOG_FACILITY=") + DECIMAL_STR_MAX(int)];
+             syslog_facility[sizeof("SYSLOG_FACILITY=") + DECIMAL_STR_MAX(int)], *msg;
         const char *message = NULL, *syslog_identifier = NULL, *syslog_pid = NULL;
         struct iovec iovec[N_IOVEC_META_FIELDS + 6];
-        unsigned n = 0;
+        unsigned n = 0, i;
         int priority = LOG_USER | LOG_INFO;
         _cleanup_free_ char *identifier = NULL, *pid = NULL;
-        const char *orig;
 
         assert(s);
         assert(buf);
 
-        orig = buf;
-        syslog_parse_priority(&buf, &priority, true);
+        /* We are creating copy of the message because we want to forward original message verbatim to the legacy
+           syslog implementation */
+        for (i = buf_len; i > 0; i--)
+                if (!strchr(WHITESPACE, buf[i-1]))
+                        break;
+
+        msg = newa(char, i + 1);
+        *((char *) mempcpy(msg, buf, i)) = 0;
```

Comment:
This assignment and cast seem very unnecessary. Why are they there?

Reply:
This was adviced by Lennart as an optimization. My previous version called ```newa0``` and ```memcpy```. Of course, I agree that this is extremely ugly.

Reply:
Right, I see now.